app/services/dataAccess/InMemoryAccessObject.java

```diff
@@ -1,58 +1,128 @@
 package services.dataAccess;
 
-import services.dataAccess.AbstractDataAccess;
-import services.dataAccess.proto.PostListProto;
-import services.dataAccess.proto.PostProto;
+import services.dataAccess.proto.PostListProto.PostList;
+import services.dataAccess.proto.PostProto.Post;
 
-import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.*;
 
 /**
  * Created by erik on 23/10/16.
  */
 public class InMemoryAccessObject extends AbstractDataAccess {
 
-    private HashMap<byte[], ArrayList<byte[]>> dataStore;
+    private Map<String, List<Post>> postDataStore;
+    private Map<String, List<PostList>> postListDataStore;
 
     public InMemoryAccessObject() {
-        dataStore = new HashMap<>();
+        postDataStore = new HashMap<>();
+        postListDataStore = new HashMap<>();
     }
 
-    public long addNewPost(String keyString, PostProto.Post post) {
-        ArrayList<byte[]> listAtKeyString = dataStore.get(keyString.getBytes());
-        listAtKeyString.add(post.toByteArray());
+    @Override
+    public long addNewPost(String keyString, Post post) {
+
+        List<Post> listAtKeyString;
+
+        // If key exists in map, append post to list,
+        if (postDataStore.containsKey(keyString)) {
+            listAtKeyString = postDataStore.get(keyString);
+            listAtKeyString.add(post);
+
+            // otherwise, create new map entry containing post in list
+        } else {
+            listAtKeyString = new ArrayList<>();
+            listAtKeyString.add(post);
+            postDataStore.put(keyString, listAtKeyString);
+        }
         return listAtKeyString.size();
     }
 
-    public long addNewPostList(String keyString, PostListProto.PostList postList) {
-        ArrayList<byte[]> listAtKeyString = dataStore.get(keyString.getBytes());
-        listAtKeyString.add(postList.toByteArray());
+    @Override
+    public long addNewPosts(String keyString, List<Post> listOfPosts) {
+
+        List<Post> listAtKeyString;
+
+        // If key exists in map, append post to list
+        if (postDataStore.containsKey(keyString)) {
+            listAtKeyString = postDataStore.get(keyString);
+            listAtKeyString.addAll(listOfPosts);
+
+        } else {
+            // otherwise initialize key
+            listAtKeyString = postDataStore.put(keyString, new ArrayList<>(listOfPosts));
```

Comment:
deep copy needed to store unique list without it changing down the line when listOfPosts is changed, as discussed.